pkg/distributor/distributor.go

```diff
@@ -351,7 +329,46 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		return d.ingestionRate.Rate()
 	})
 
-	d.forwarder = forwarding.NewForwarder(reg, d.cfg.Forwarding)
+	// Create the configured ingestion rate limit strategy (local or global). In case
+	// it's an internal dependency and can't join the distributors ring, we skip rate
+	// limiting.
+	var ingestionRateStrategy, requestRateStrategy limiter.RateLimiterStrategy
+	var distributorsLifeCycler *ring.BasicLifecycler
```

Comment:
Nit: can we use `Lifecycler` as single word (instead of `LifeCycler`) everywhere? Just for consistency with its spelling in `ring`